core-web/src/main/java/org/silverpeas/core/web/mvc/controller/SilverpeasWebUtil.java

```diff
@@ -146,4 +156,26 @@ public String getContentLanguage(HttpServletRequest request) {
     }
     return contentLanguage;
   }
+
+  /**
+   * Gets the user favorite language from elements of the given request.
+   * @param request an instance of {@link HttpServletRequest}.
+   * @return a string representing the favorite language of the user behind the session or the
+   * language of WEB browser (or default language).
+   */
+  public String getUserLanguage(HttpServletRequest request) {
+    return Optional.ofNullable(getMainSessionController(request))
+        .filter(c -> !User.getCurrentRequester().isAnonymous())
+        .map(MainSessionController::getFavoriteLanguage)
```

Comment:
My first intention here was to centralize the getting of user language into SilverpeasWebUtil.
After that, my first implementation was exactly the one you wrote above. But this implementation was induicing performance problems. Each time `u.getUserPreferences().getLanguage()` is performed, each time a SQL query is performed.
I had no time to spend on this point and for now the language is taken from the session.

Reply:
Ok, to avoid an SQL request each time the user preferences are asked, they have just to be declared as an attribute of the `UserDetails` class. I just did this, you can synch your basecode with master and then update the `SilverpeasWebUtil#getUserLanguage(HttpServletRequest)` method